### DISCUSSION TITLE
Possible to create new engine from CXPatcher patches?  (similar to D3DMetal)

### DISCUSSION THREAD

**Discussion by sofakng** [The-Wineskin-Project/WineskinServer#445] [General]
It looks like CXPatcher contains additional patches (ie. D3D11 geometry shaders, etc) and possibly other stuff not included in the latest Wineskin engine.

Is it possible to include them and build a new engine similar to the instructions on adding D3DMetal?

It seems easy enough to update the engine's wswine.bundle but I'm not sure which files would need to be copied...

**Comment by Gcenx** [comment]
CXPatcher does _not_ provide geometry shaders, instead it uses a slightly tweaked copy of MoltenVK & DXVK that workaround the lack of geometry shaders.

If you use CXPatchers copy of MoltenVK & DXVK you’ll get the same results in Wineskin.

_Why_ doesn’t Wineskin already have these features?, I’m the sole developer working on all components of Wineskin.

**Comment by sofakng** [comment]
Thanks again for the information.  I wasn't asking (and definitely not complaining) about why it wasn't included.

Part of my reason for asking so many questions is to be able to help in the future.  I can tell there are a lot of components and it seems like quite a lot for a single person.
